sycl/include/sycl/ext/oneapi/matrix/matrix-hip.hpp

```diff
@@ -0,0 +1,317 @@
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(bfloat16, b, 16, 16, 4)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(bfloat16, a, 32, 8, 4)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(bfloat16, b, 8, 32, 4)
+
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(half, a, 16, 16, 4)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(half, b, 16, 16, 4)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(half, a, 32, 8, 4)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(half, b, 8, 32, 4)
+
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(double, a, 16, 4, 1)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR(double, b, 4, 16, 1)
+
+#undef __SYCL_JOINT_MATRIX_OVERLOAD_ARR
+
+#define __SYCL_JOINT_MATRIX_OVERLOAD_INT8_ARR(USE, M, N, SIZE)                 \
+  template <matrix_layout Layout>                                              \
+  struct joint_matrix_hip<                                                     \
+      int8_t, matrix_use::USE, M, N, Layout,                                   \
+      typename std::enable_if_t<Layout == matrix_layout::row_major ||          \
+                                Layout == matrix_layout::col_major>> {         \
+    int8_t data[SIZE];                                                         \
+  };
+
+__SYCL_JOINT_MATRIX_OVERLOAD_INT8_ARR(a, 32, 8, 4)
+__SYCL_JOINT_MATRIX_OVERLOAD_INT8_ARR(b, 8, 32, 4)
+__SYCL_JOINT_MATRIX_OVERLOAD_INT8_ARR(a, 16, 16, 4)
+__SYCL_JOINT_MATRIX_OVERLOAD_INT8_ARR(b, 16, 16, 4)
+
+#undef __SYCL_JOINT_MATRIX_OVERLOAD_INT8_ARR
+
+#define __SYCL_JOINT_MATRIX_OVERLOAD_ARR_ACC(TYPE, M, N)                       \
+  template <>                                                                  \
+  struct joint_matrix_hip<TYPE, matrix_use::accumulator, M, N,                 \
+                          matrix_layout::dynamic> {                            \
+    using ext_array_t =                                                        \
+        __attribute__((__vector_size__((M * N) / 64 * sizeof(TYPE)))) TYPE;    \
+    ext_array_t data = {0};                                                    \
+  };
+
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR_ACC(float, 16, 16)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR_ACC(float, 32, 32)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR_ACC(double, 16, 16)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR_ACC(int32_t, 32, 32)
+__SYCL_JOINT_MATRIX_OVERLOAD_ARR_ACC(int32_t, 16, 16)
+
+#undef __SYCL_JOINT_MATRIX_OVERLOAD_ARR_ACC
+
+template <matrix_layout Layout, typename S, typename T, size_t NumRows,
+          size_t NumCols, access::address_space Space,
+          access::decorated IsDecorated, typename Group>
+void load_accumulator_layoutT(
+    joint_matrix_hip<S, matrix_use::accumulator, NumRows, NumCols,
+                     matrix_layout::dynamic> &res,
+    multi_ptr<T, Space, IsDecorated> src, size_t stride, Group &sg) {
+  std::ignore = stride;
+  const auto idx = sg.get_group_linear_id() * sg.get_local_range()[0] +
+                   sg.get_local_linear_id();
+  if constexpr (Layout == matrix_layout::row_major) {
+    res.data = {0};
+  } else if constexpr (Layout == matrix_layout::col_major) {
+    res.data = {0};
```

Comment:
So it seems this doesn't load any data to the accumulator? I guess you'll add this before opening a PR?